docs/proposals/bulward-integration.md

```diff
@@ -0,0 +1,206 @@
+This proposal is about how we integrate [Bulward - Multi Projects/Users Management](https://github.com/kubermatic/bulward)
+project to KubeCarrier. This is because bulward has more advanced RBAC management features, and we can offload permission
+handling completely from KubeCarrier.
+## Objects Reorganization
+### KubeCarrier Account
+In KubeCarrier, the only object for user representation is `Account`. Here is an example of what `Account` object looks like:
+```yaml
+apiVersion: catalog.kubecarrier.io/v1alpha1
+kind: Account
+metadata:
+  name: team-a
+spec:
+  metadata:
+    displayName: The A Team
+    shortDescription: In 1972, a crack commando unit was sent to prison by a military court...
+  roles:
+  - Provider
+  - Tenant
+  subjects:
+  - kind: User
+    name: hannibal
+    apiGroup: rbac.authorization.k8s.io
+  - kind: User
+    name: team-a-member
+    apiGroup: rbac.authorization.k8s.io
+```
+`roles` in the `Account` indecates if the users in `Account` are service providers or service consumers (tenants). In
+KubeCarrier ServiceHub, `Account` can be both `Provider` and `Tenant` at the same time, which means user can consume services
+via KubeCarrier ServiceHub, at the same time, user can also provide the services to other users.
+`Account` is a **cluster-scoped** object, and each `Account` has a dedicated namespace.
+### Bulward Organization & Project
+In bulward, there are two objects for user representation.
+One is `Organization`, which is cluster-scoped:
+```yaml
+apiVersion: apiserver.bulward.io/v1alpha1
+kind: Organization
+metadata:
+  name: organization-a
+spec:
+  metadata:
+    displayName: The A Organization
+    description: I am the first orgainzation of Bulward.
+  owners:
+  - kind: User
+    name: owner
+    apiGroup: rbac.authorization.k8s.io
+```
+And each `Organization` has a namespace, Organization owner or the user who has permission to create `Project` objects can
+create `Project` object in the `Organization` namespace:
+```yaml
+apiVersion: apiserver.bulward.io/v1alpha1
+kind: Project
+metadata:
+  name: project-example
+  namespace: organization-a
+spec:
+  owners:
+  - kind: User
+    name: owner
+    apiGroup: rbac.authorization.k8s.io
+```
+`Project` is namespace-scoped, and similar to `Organization`, every `Project` will have a namespace.
+
+### Account <-> Organization or Project?
+There are few reasons that we don't replace the `Account` with `Project`:
+- KubeCarrier cannot take fully advantage of Organization features in Bulward. For big enterprises, `Organzation` concept
+is needed for them to manage permissions of different team, different user has different permission for different project.
+- `Account` is cluster-scoped object, while `Project` is namespace-scoped object, integration will introduce lot of changes.
+
+So this proposal proposes the following changes:
+We use `Organziation` to represent the `Account` in KubeCarrier. For better explanation, here I will provide an example of
```

Comment:
This is kinda a moot point. I'd argue most `Accounts` would be either pure tenants or pure providers. There would be a minority being both a tenant and a provider, which IMHO would be providers with service dependencies.